package/MDAnalysis/coordinates/base.py

```diff
@@ -980,6 +980,78 @@ def __repr__(self):
             nframes=self.n_frames,
             natoms=self.n_atoms
         ))
+    
+    def timeseries(self, asel: Optional['AtomGroup']=None,
+                   start: Optional[int]=None, stop: Optional[int]=None,
+                   step: Optional[int]=None,
+                   order: Optional[str]='fac') -> np.ndarray:
+        """Return a subset of coordinate data for an AtomGroup
+
+        Parameters
+        ----------
+        asel : AtomGroup (optional)
+            The :class:`~MDAnalysis.core.groups.AtomGroup` to read the
+            coordinates from. Defaults to ``None``, in which case the full set
+            of coordinate data is returned.
+        start :  int (optional)
+            Begin reading the trajectory at frame index `start` (where 0 is the
+            index of the first frame in the trajectory); the default
+            ``None`` starts at the beginning.
+        stop : int (optional)
+            End reading the trajectory at frame index `stop`-1, i.e, `stop` is
+            excluded. The trajectory is read to the end with the default
+            ``None``.
+        step : int (optional)
+            Step size for reading; the default ``None`` is equivalent to 1 and
+            means to read every frame.
+        order : str (optional)
+            the order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+
+        See Also
+        --------
+        :class:`MDAnalysis.coordinates.memory`
+
+
+        .. versionadded:: 2.4.0
+        """
+        start, stop, step = self.check_slice_indices(start, stop, step)
+        nframes = len(range(start, stop, step))
+
+        if asel is not None:
+            if len(asel) == 0:
+                raise NoDataError(
```

Comment:
As per #3898, I believe both @orbeckst and @richardjgowers were keen to enforce that `NoDataError` was a topology attrs thing rather than zero width atomgroups (fwiw I also thought it was meant for these cases, but I'm happy going with the alternative).

Would we be ok just raising a ValueError here and then switching the DCDReader to ValueError? I don't necessarily think we need to semver this change, I doubt anyone is specifically looking for this specific edge case anywhere?

Reply:
I agree with making it `ValueError`.